### DISCUSSION TITLE
Question: Cannot build, unexpected &

### DISCUSSION THREAD

**Discussion by gameboycjp** [FunkinCrew/Funkin#1480] [Compiling help]
I'm trying to build the game on Debian, but can't get it working. I get a syntax error from the newgrounds lib.
/home/gameboycjp/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/objects/events/Result.hx:32: characters 35-36 : Unexpected &

Did I do something wrong with the setup?

**Comment by ghost** [comment]
I just tried it myself and built it from source on ubuntu
and just only got these warnings no syntax error
command used:
`haxelib run lime test linux -debug`

Warnings:
```
/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:201: characters 31-37 : 
Warning : This typedef is deprecated in favor of sys.thread.Thread

/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:72: characters 14-20 : 
Warning : This typedef is deprecated in favor of sys.thread.Thread

/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:77: characters 13-19 : 
Warning : Usage of this typedef is deprecated

/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:183: characters 31-37 : 
Warning : Usage of this typedef is deprecated

/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:174: characters 32-38 : 
Warning : Usage of this typedef is deprecated

/home/itami/haxelib/newgrounds/1,1,3/lib/Source/io/newgrounds/utils/AsyncHttp.hx:96: characters 32-38 : 
Warning : Usage of this typedef is deprecated

/home/itami/haxelib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : 
Warning : This typedef is deprecated in favor of haxe.xml.Access

/home/itami/haxelib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : 
Warning : This typedef is deprecated in favor of haxe.xml.Access

source/TitleState.hx:61: characters 16-28 : Type not found : APIStuff
```

**Comment by regulad** [comment]
Uh, looks like you need to make `source/APIStuff.hx`

**Comment by gameboycjp** [comment]
@regulad I already have. Copied text inside and all.

**Comment by edlrod** [comment]
Same problem compiling to html5

**Comment by ghost** [comment]
> @regulad I already have. Copied text inside and all.

that was a reply to my comment since it says that APIStuff was not found but its weird that you get a different error than me

**Comment by cow03haha** [comment]
Is that your haxe version is correct? I have same problem and i solve it by upgrade haxe 3.4.7(debian buster stable package) to required version

**Comment by ghost** [comment]
i used Haxe 4.2.1 and it works fine

**Comment by rodrigo1406** [comment]
Got the same problem here. I have haxe 3.4.7 and haxelib 3.3.0, using Debian Buster. Trying to compile from the source I've just downloaded.

lime test html5 -debug
/home/rodrigo/haxelib/newgrounds/1,1,4/lib/Source/io/newgrounds/objects/events/Result.hx:32: characters 35-36 : Unexpected &

Then I try to continue, as if it's nothing serious, but got another error:

lime test linux -debug
source/Discord.hx:58: characters 48-49 : Unexpected )
source/Discord.hx:58: characters 48-49 : Unexpected )

In the end there's no "linux" directory inside export/release/ (only a .build file).

Another thing, in the "Ignored files" section of the manual, there's a line:

"Just make a file in /source and call it APIStuff.hx, and copy paste this into it"

the correct would be source/ instead of /source (which indicates a "source" folder in the root / directory).

**Comment by ghost** [comment]
You require Haxe 4.1.5 or 4.2.1/later

**Comment by Ph4ntomShad0w** [comment]
well my error is source/APIStuff.hx:1: characters 1-5 : Unexpected haxe
how do i fix it?

**Comment by ghost** [comment]
```haxe
package;

class APIStuff
{
	public static var API:String = "";
	public static var EncKey:String = "";
}
```
Paste that in

**Comment by zoom1220** [comment]
Also having this same issue on Windows.

**Comment by ghost** [comment]
> Got the same problem here. I have haxe 3.4.7 and haxelib 3.3.0, using Debian Buster. Trying to compile from the source I've just downloaded.
> 
> lime test html5 -debug
> /home/rodrigo/haxelib/newgrounds/1,1,4/lib/Source/io/newgrounds/objects/events/Result.hx:32: characters 35-36 : Unexpected &
> 
> Then I try to continue, as if it's nothing serious, but got another error:
> 
> lime test linux -debug
> source/Discord.hx:58: characters 48-49 : Unexpected )
> source/Discord.hx:58: characters 48-49 : Unexpected )
> 
> In the end there's no "linux" directory inside export/release/ (only a .build file).
> 
> Another thing, in the "Ignored files" section of the manual, there's a line:
> 
> "Just make a file in /source and call it APIStuff.hx, and copy paste this into it"
> 
> the correct would be source/ instead of /source (which indicates a "source" folder in the root / directory).

Haxe 4.1.4 or earlier won't work, neither will Haxe 4.2.0. Add the Haxe PPA.

**Comment by h8games87023** [comment]
i also have this error source/Controls.hx:411: characters 25-33 : Array<Int> has no field contains
/home/augustin/augustin/Desktop/polymod/git/polymod/fs/PolymodFileSystem.hx:19: characters 16-18 : Expected )
 im on linux and i suck so can anyone kindly help me?
